### ISSUE TITLE
There may be a little mistake in main.py?

### ISSUE THREAD

**Issue by superpig99** [linyongver/ZIN_official#1]
Hi guru,
The 100th line of the main.py file seems to be written like this: `train_penalty = torch.max(torch.tensor(0.0).cuda(), train_penalty.detach().cuda())`, otherwise, the 106th line will report the following error: 
![image](https://user-images.githubusercontent.com/91784277/208121671-4f3d61d9-b050-4a34-8b2c-47be7e12f159.png)

Please correct me if I am wrong :)

**Comment by zhushyu** [comment]
> Hi guru, The 100th line of the main.py file seems to be written like this: `train_penalty = torch.max(torch.tensor(0.0).cuda(), train_penalty.detach().cuda())`, otherwise, the 106th line will report the following error: ![image](https://user-images.githubusercontent.com/91784277/208121671-4f3d61d9-b050-4a34-8b2c-47be7e12f159.png)
> 
> Please correct me if I am wrong :)

Hi, thanks for reaching out. We've got another question regarding the implementation, which we believe is due to different Pytorch verions. Can you let us know what Pytorch version you are using? We will check if there is other similar issues under this version. 

Thanks again.

**Comment by superpig99** [comment]
> 

The version of Pytorch I'm using is 1.7.1+cu110.

**Comment by aamixsh** [comment]
I was trying this for my own dataset and am getting the same error. Not sure if what @superpig99 mentioned is the cause of the problem though. 

I think there is a logical inconsistency in **infer_irmv1.py** where the `infer_env` model is supposed to be updated by a backward on `-train_penalty`. However, `train_penalty` is also a part of the main model and that makes the main model's `loss.backward()` fail because that has been updated already to a newer version (an in-place operation probably). When I comment out this `infer_env` model update, the code runs but obviously wouldn't work without that update. 

Is my understanding correct? And how can we fix this error?
<img width="498" alt="image" src="https://user-images.githubusercontent.com/19327710/213309639-cc962e9e-6b84-4796-9802-35789355b4f2.png">

**Comment by hututu2** [comment]
Hello, I also encountered the same problem. I solved it by changing the versions of python and torch. The environment I can run is: python 3.10+torch 2.0 (windows 11+cuda 11.8).
